src/python/pants/base/revision.py

```diff
@@ -93,6 +93,18 @@ def components(self):
   def _is_valid_operand(self, other):
     return hasattr(other, '_components')
 
+  def _fill_value_if_missing(self, ours, theirs):
```

Comment:
Python 2 allows comparing ints and strs, but Py3 doesn't. So, the old fillvalue of `0` for missing values would not work. Instead, it should be `0` if dealing with ints and `''` if dealing with strs. 

`__lt__` was changed to have a fillvalue of `None`, which is then used in this helper function to determine if it should be a str or int value.